src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -164,6 +183,21 @@ private enum Flags : byte
             }
         }
 
+        internal void CreateBackingField(bool isCreatedForFieldKeyword)
+        {
+            lock (_backingFieldLock)
```

Comment:
>lock (_backingFieldLock) [](http://example.com/codeflow?start=12&length=24)

We usually use ```InterlockedOperations.Initialize``` or ```Interlocked.CompareExchange``` API for initializations like that. This has less overhead and doesn't require a lock object.